SimPEG/regularization/base.py

```diff
@@ -373,34 +486,100 @@ def deriv2(self, m, v=None) -> csr_matrix:
 
 
 class Smallness(BaseRegularization):
-    r"""
-    Small regularization - L2 regularization on the difference between a
-    model and a reference model.
+    r"""Smallness regularization for least-squares inversion.
+
+    ``Smallness`` regularization is used to ensure that differences between the
+    model values in the recovered model and the reference model are small;
+    i.e. it preserves structures in the reference model. If a reference model is not
+    supplied, the starting model will be set as the reference model in the
+    corresponding objective function by default. Optionally, custom cell weights can be
+    included to control the degree of smallness being enforced throughout different
+    regions the model.
+
+    See the *Notes* section below for a comprehensive description.
+    
+    Parameters
+    ----------
+    mesh : .regularization.RegularizationMesh
+        Mesh on which the regularization is discretized. Not the mesh used to
+        define the simulation.
+    active_cells : None, (n_cells, ) numpy.ndarray of bool
+        Boolean array defining the set of :py:class:`~.regularization.RegularizationMesh`
+        cells that are active in the inversion. If ``None``, all cells are active.
+    mapping : None, SimPEG.maps.BaseMap
+        The mapping from the model parameters to the active cells in the inversion.
+        If ``None``, the mapping is the identity map.
+    reference_model : None, (n_param, ) numpy.ndarray
+        Reference model. If ``None``, the reference model in the inversion is set to
+        the starting model.
+    units : None, str
+        Units for the model parameters. Some regularization classes behave
+        differently depending on the units; e.g. 'radian'.
+    weights : None, dict
+        Weight multipliers to customize the least-squares function. Each key points to
+        a (n_cells, ) numpy.ndarray that is defined on the
+        :py:class:`regularization.RegularizationMesh` .
```

Comment:
I agree. For now, I was told to duplicate the input arguments most likely to be set by the user. I think Joe is working on something to avoid this issue.